### PR TITLE
EmberAddon Should Merge Defaults

### LINKED ISSUE #4435: ember-cli-build.js template for addons doesn't allow passing options
The default template for `ember-cli-build.js` in an addon looks something like:

```js
/* global require, module */
var EmberApp = require('ember-cli/lib/broccoli/ember-addon');

module.exports = function(defaults) {
  var app = new EmberApp(defaults, {
    // Add options here
  });

  /*
    This build file specifes the options for the dummy test app of this
    addon, located in `/tests/dummy`
    This build file does *not* influence how the addon or the app using it
    behave. You most likely want to be modifying `./index.js` or app's build file
  */

  return app.toTree();
};
```

However, `ember-cli/lib/broccoli/ember-addon` expects only one argument (the `options`) so options put in `// Add options here` don't get successfully passed into the app.

### PR DESCRIPTION
**Pull request by chadhietala**
This closes #4435.

### CODE REVIEW

File: lib/broccoli/ember-addon.js [hunk 1]
```diff
@@ -19,8 +19,20 @@ module.exports = EmberAddon;
...
+  var options = {};
+
+  for (var i = 0, l = arguments.length; i < l; i++) {
+    args.push(arguments[i]);
+  }
+
+  if (args.length === 1) {
+    options = args[0];
+  } else if (args.length > 1) {
+    options = defaults.apply(null, args);
```
Comment (stefanpenner) [inline lib/broccoli/ember-addon.js]: does defaults merge multiple levels? (is that needed?)

**Comment by stefanpenner** [comment]
seems like once this lands we need to cut a new release.

File: lib/broccoli/ember-addon.js (hunk 1 above)
Comment (rwjblue) [inline lib/broccoli/ember-addon.js]: We use a slightly different pattern in ember-app for deeply merged values ([see here](https://github.com/ember-cli/ember-cli/blob/master/lib/broccoli/ember-app.js#L199-L212)), which is something like:

```javascript
var defaults      = require('lodash/object/defaults');
var merge         = require('lodash/object/merge');

var defaults      = require('lodash/object/defaults');
var merge         = require('lodash/object/merge');

options = merge({}, args, defaults);
```

**Comment by chadhietala** [comment]
@stefanpenner @rwjblue Added `merge-defaults` which will recursively merge defaults.

**Comment by stefanpenner** [comment]
kicking travis

**Comment by stefanpenner** [comment]
@chadhietala awesome!

**Comment by rwjblue** [comment]
Can this PR also address #4434 (or maybe it does)?
Can we have a test to ensure we don't regress in the future?

**Comment by chadhietala** [comment]
@rwjblue addresses #4434. Added tests.

**Comment by stefanpenner** [comment]
looks correct.

**Comment by chadhietala** [comment]
Added a third level array.

**Comment by stefanpenner** [comment]
> Added a third level array.

thanks, :shipit: when green.

**Comment by rwjblue** [comment]
Thank you!

**Pull request merged**